PyInstaller/building/osx.py

```diff
@@ -146,6 +152,24 @@ def assemble(self):
 
                            }
 
+        if self.osx_app_console:
+            # Make app bundle double-clickable
+            app_path = Path(self.name)
+
+            info_plist_dict['CFBundleExecutable'] = 'wrapper'
+
+            # write new wrapper script
+            shell_script = '''#!/bin/bash
+            dir=$(cd "$( dirname "${0}")" && pwd )
+            open -a Terminal "file://${dir}/%s"''' % shlex.quote(self.appname)
```

Comment:
It turns out this `shlex.quote()` call is redundant (because the string is already in double quotes), and is actually harmful because if `self.appname` contains a space, it will add single quotes around the name. So we end up with `open -a Terminal "file://${dir}/'program with space'"` (i.e., single quotes within double quotes), which leads to an error.

And it also seems that if `file://` uri schema is used, the spaces need to be url-encoded (i.e., %20), otherwise we get an error as well. I'm not sure if url-encoding from bash is feasible, though (we could url-encode `self.appname` on python side, but if self.appname contains a space, so will ${dir} on the script side...). Removing the `file://` schema seems to make non-encoded spaces work...

Reply:
so removing `file://` and `shelex.quote` seems to fix things? I will try that